api/v1alpha1/k8ssandracluster_types.go

```diff
@@ -76,42 +76,77 @@ type K8ssandraClusterList struct {
 	Items           []K8ssandraCluster `json:"items"`
 }
 
-type Cassandra struct {
+type CassandraClusterTemplate struct {
+	// Cluster is the name of the cluster. This corresponds to cluster_name in
+	// cassandra.yaml.
+	// +kubebuilder:validation:MinLength=2
 	Cluster string `json:"cluster,omitempty"`
 
-	Datacenters []CassandraDatacenterTemplateSpec `json:"datacenters,omitempty"`
+	// ServerImage is the image for the cassandra container. Note that this should be a
+	// management-api image. If left empty the operator will choose a default image based
+	// on ServerVersion.
+	ServerImage string `json:"serverImage,omitempty"`
+
+	// ServerVersion is the Cassandra version.
+	// +kubebuilder:validation:Pattern=(3\.11\.\d+)|(4\.0\.\d+)
+	ServerVersion string `json:"serverVersion,omitempty"`
+
+	// Resources is the cpu and memory resources for the cassandra container.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// SystemLoggerResources is the cpu and memory resources for the server-system-logger
+	// container.
+	SystemLoggerResources *corev1.ResourceRequirements `json:"systemLoggerResources,omitempty"`
```

Comment:
Is this really something we want to make so configurable? Are we going to expose all the init containers and their resource configurations also?

Reply:
Good question. When I started on this PR I picked some arbitrary properties to expose in order to help me get started with the code changes. I definitely think we can avoid exposing the system logger resources and probably for server-config-init as well. 

I do think we will need to expose configs for other containers like for backup/restore.

Reply:
I am going to leave this as is for now. I created #120 to come back and address. If there is consensus to remove that's fine with me.